### PR TITLE
Add skipping daiy digest when there are only tasks due after today

### PR DESCRIPTION
**Pull request by gbochora**
#### Summary
This PR excludes summary info from daily digest message when there are only tasks due in the future and returns empty text. As a result, the daily digest message is skipped when we only have tasks due in the future.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43586

#### Checklist

- ~~[ ] Telemetry updated~~
- ~~[ ] Gated by experimental feature flag~~
- [x] Unit tests updated

### CODE REVIEW

File: server/app/playbook_run_service.go [hunk 1]
```diff
@@ -2829,17 +2829,22 @@ func buildAssignedTaskMessageSummary(runs []AssignedRun, locale string, timezone
...
-		}
-		msg.WriteString("\n\n")
-		msg.WriteString(runsInfo.String())
+	// if there are only tasks that are due after today and we need to show only tasks due now, skip a message
+	if onlyDueUntilToday && tasksDoAfterToday == total {
+		return ""
 	}
 
+	// add title
+	if onlyDueUntilToday {
```
Comment (crspeller) [inline server/app/playbook_run_service.go:2838]: Logic is a bit hard to follow here. We seem to have gotten rid of the behavior where if the `runsInfo` is empty then we skip writing this stuff. Was that intentional?

**Review by crspeller — CHANGES_REQUESTED** [review]

File: server/app/playbook_run_service.go (hunk 1 above)
Comment (gbochora) [reply server/app/playbook_run_service.go:2838]: Yep, it's pretty complicated. 
Instead of `runsInfo` is empty I've added `tasksDoAfterToday == total` condition, lines 2832-2835. 
The logic here is: If we need only tasks that are due until today and all tasks are due after today(`tasksDoAfterToday == total`), we return an empty message.

**Review by agarciamontoro — APPROVED** [review]
Looks good!

**Review by crspeller — APPROVED** [review]
Looks good!
